### ISSUE TITLE
Add Native American Reservation Boundary as an optional clickable layer/overlay

### ISSUE THREAD

**Issue by Makosak** [GeoDaCenter/covid#39]
Requested by Indian Health Service

**Comment by Makosak** [comment]
Could we do this by next week? Someone could identify the shapefile and attach here; a JS volunteer could add?

**Comment by rbrtmrtn** [comment]
I found a layer for this on ArcGIS Online which looks pretty recent (last updated 2019). 

https://www.arcgis.com/home/item.html?id=108d7cbecbca42549c49b1454bfa7c9c

You can pull all the features as GeoJSON with this URL:

https://services4.arcgis.com/ul3DV3LWHhAM8P61/arcgis/rest/services/Navajo_Nation_2013/FeatureServer/0/query?f=geojson&where=1%3D1&returnGeometry=true&outSR=4326

I can drop that into the map, that shouldn't be an issue.

**Comment by rbrtmrtn** [comment]
Just a quick update on this—I was able to bring the layer in but it seems to be broken up into communities:

![image](https://user-images.githubusercontent.com/1463553/79139068-3f9e2680-7d7b-11ea-9ad8-5a8e26e401e0.png)

Looking at the [metadata](https://www.arcgis.com/home/item.html?id=108d7cbecbca42549c49b1454bfa7c9c) this seems about right:

> Shapfile completed with following references:
> 1. 1978 Census Community Boundaries established by Navajo Nation Court of Appeals (Case No. A-CV-19-77) March 17, 1978
> 2. Bureau of Indian Affairs Grazing Units for Eatern Navajo Agency
> 3. Division of Community Development's Chapter Images (Chapter Profiles 2004)

@Makosak I was going to dissolve the polygons next but wanted to check in with you (and perhaps the stakeholders) first! Thank you!

**Comment by rbrtmrtn** [comment]
Just a quick update—I was able to add the boundaries but didn't realize they're broken down by community:

![image](https://user-images.githubusercontent.com/1463553/79142145-a70aa500-7d80-11ea-89fc-41e554f528e0.png)

@Makosak should I go ahead merge these polygons? Also, there's a live demo here if anyone would like to see the data as-is :) 

https://covid-atlas-rbrtmrtn.netlify.com/map.html

Thanks guys!

**Comment by rbrtmrtn** [comment]
There's a live demo here if anyone would like to see the data as-is:

https://covid-atlas-rbrtmrtn.netlify.com/map.html

**Comment by Makosak** [comment]
Looking good to start! I like the demo version.

We shouldn't jut include the Navajo Nation (that was just an example of why
we need this), but all the First Nation reservation boundaries. What about
this file?
https://hifld-geoplatform.opendata.arcgis.com/datasets/54cb67feef5746e8ac7c4ab467c8ae64

I think a slightly transparent dark gray boundary may work better? Should
we try with a light label for each or will that get to be too much?

**Comment by rbrtmrtn** [comment]
Sorry about that @Makosak, I'm not sure where I got the idea it was just the Navajo Nation 🤔 I can definitely add the Indian Reservations boundary—will update you soon!

**Comment by rbrtmrtn** [comment]
I downloaded the reservations shapefile @Makosak and converted it to GeoJSON but it's quite big (23 MB) 😕 I think we may want to put this one into Mapbox. Does the Center have an account or would it be OK if I created one?

**Comment by rbrtmrtn** [comment]
Reservations GeoJSON has been uploaded as a Mapbox tileset here:

https://studio.mapbox.com/tilesets/lixun910.4j9fklls/

**Comment by rbrtmrtn** [comment]
Deck.gl has a generic `TileLayer` component for adding tiled data, but it doesn't know how to consume Mapbox vector tiles by default. There's a recipe for that here:

https://tgorkin.github.io/docs/layers/tile-layer

This relies on two npm packages that aren't available as browser builds (`@mapbox/vector-tiles` and `@mapbox/point-geometry`), so I cribbed them into this project and degraded them from CommonJS to vanilla.

The good news is I did get the reservations to render in Deck:

![image](https://user-images.githubusercontent.com/1463553/80793838-d3cf0280-8b5d-11ea-990b-5a91ec446dff.png)

One thing that will require more finessing is that the reservations aren't visible when you first load them—you have to zoom in a little. The is because Mapbox tries to intuit reasonable min/max zoom levels when you upload data, and their min is beyond our initial zoom level. According to their docs, the only way to adjust zoom constraints is to build the tiles yourself locally using Tippecanoe, and then uploading the `.mbtiles` file it spits out.

https://docs.mapbox.com/help/troubleshooting/adjust-tileset-zoom-extent/#transform-data-with-tippecanoe

It seems like we'll have to do this first, and then try to understand how to style layers in Deck. The documentation around that doesn't appear to be exhaustive, but hopefully it's enough to point us in the right direction:

https://github.com/visgl/deck.gl/blob/master/docs/api-reference/layer.md

**Comment by rbrtmrtn** [comment]
Tippecanoe command for generating MBTiles at inferred min/max zoom levels, for posterity 😄 

```
tippecanoe -o native-american-reservations.mbtiles -zg native-american-reservations.geojson
```

**Comment by rbrtmrtn** [comment]
A few notes from @lixun910 on how the reservation boundaries are surfacing an issue with the zoom controls:

> After click on the zoom button, show native american reservations checkbox won’t work anymore.

Followup:

> I just checked that the zoom button also doesn’t work with other maps on the original atlas, I think probably it is a bug of the hacking style zoom button, that needs to be fixed. Thanks!

Thank you again Xun for letting us know about this 😄

**Comment by rbrtmrtn** [comment]
Closed by d4766b6c252588bec885c3621b728b6c86617bfb
